File: frontend/lib/models/message.dart
class Message {
  final int id;
  final int sender;
  final int receiver;
  final int chatId;
  final String messageText;
  final String timeSent;

  Message(
      {required this.id,
      required this.sender,
      required this.receiver,
      required this.chatId,
      required this.messageText,
      required this.timeSent});

  factory Message.fromJson(Map<String, dynamic> json) {
    return Message(
        id: json['id'],
        sender: json['sender'],
        receiver: json['receiver'],
        chatId: json['chatId'],
        messageText: json['messageText'],
        timeSent: json['timeSent']);
  }
}

File: backend/src/main/java/com/ceniuch/backend/Entities/Message.java
package com.ceniuch.backend.Entities;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;

import java.sql.Timestamp;

@Entity
public class Message {
    @Id
    int id;
    int sender;
    int receiver;
    int chatId;
    String messageText;
    Timestamp timeSent;

    public Message() {

    }

    public Message(int id, int sender, int receiver, int chatId,
                   String messageText, Timestamp timeSent) {
        this.id = id;
        this.sender = sender;
        this.receiver = receiver;
        this.chatId = chatId;
        this.messageText = messageText;
        this.timeSent = timeSent;
    }

    public int getId() {
        return id;
    }

    public int getSender() {
        return sender;
    }

    public int getReceiver() {
        return receiver;
    }

    public int getChatId() {
        return chatId;
    }

    public String getMessageText() {
        return messageText;
    }

    public Timestamp getTimeSent() {
        return timeSent;
    }

    public void setTimeSent(Timestamp timeSent) {
        this.timeSent = timeSent;
    }
}
